cpp/src/copying/scatter.cu

```diff
@@ -164,41 +170,74 @@ struct column_scalar_scatterer_impl<string_view, MapIterator> {
 };
 
 template <typename MapIterator>
-struct column_scalar_scatterer_impl<dictionary32, MapIterator> {
+struct column_scalar_scatterer_impl<list_view, MapIterator> {
   std::unique_ptr<column> operator()(std::unique_ptr<scalar> const& source,
                                      MapIterator scatter_iter,
                                      size_type scatter_rows,
                                      column_view const& target,
                                      rmm::mr::device_memory_resource* mr,
                                      cudaStream_t stream) const
   {
-    CUDF_FAIL("scatter scalar to dictionary not implemented");
+    CUDF_FAIL("scatter scalar to list_view not implemented");
   }
 };
 
 template <typename MapIterator>
-struct column_scalar_scatterer_impl<list_view, MapIterator> {
+struct column_scalar_scatterer_impl<struct_view, MapIterator> {
   std::unique_ptr<column> operator()(std::unique_ptr<scalar> const& source,
                                      MapIterator scatter_iter,
                                      size_type scatter_rows,
                                      column_view const& target,
                                      rmm::mr::device_memory_resource* mr,
                                      cudaStream_t stream) const
   {
-    CUDF_FAIL("scatter scalar to list_view not implemented");
+    CUDF_FAIL("scatter scalar to struct_view not implemented");
   }
 };
 
 template <typename MapIterator>
-struct column_scalar_scatterer_impl<struct_view, MapIterator> {
+struct column_scalar_scatterer_impl<dictionary32, MapIterator> {
   std::unique_ptr<column> operator()(std::unique_ptr<scalar> const& source,
                                      MapIterator scatter_iter,
                                      size_type scatter_rows,
                                      column_view const& target,
                                      rmm::mr::device_memory_resource* mr,
                                      cudaStream_t stream) const
   {
-    CUDF_FAIL("scatter scalar to struct_view not implemented");
+    auto dict_target = dictionary::detail::add_keys(
+      dictionary_column_view(target),
+      make_column_from_scalar(*source, 1, rmm::mr::get_current_device_resource(), stream)->view(),
+      mr,
+      stream);
+    auto dict_view    = dictionary_column_view(dict_target->view());
+    auto scalar_index = dictionary::detail::get_index(
+      dict_view, *source, rmm::mr::get_current_device_resource(), stream);
+    auto scalar_iter = thrust::make_permutation_iterator(
```

Comment:
Why not just a constant_iterator with the scalar value? The scalar will always be int32_t so we aren't getting the indexalator compile time savings.

Reply:
The scalar type returned here can be any unsigned integer type.

Reply:
I think I'm misunderstanding some code. `get_index` only returns `numeric_scalar<int32_t>`. And this is a constant iterator that gets that value and is used in the subsequent thrust call? Or do you mean in the future when dictionary indices could be any type?

Reply:
Looks like your code is out of date. Dictionary indices can be any unsigned type right now. And `get_index` now returns a `scalar`:
https://github.com/rapidsai/cudf/blob/7081ae4d3c896a137a8e4881192884010668a3c1/cpp/include/cudf/dictionary/search.hpp#L39-L42